lib/runner.js

```diff
@@ -441,6 +442,47 @@ Runner.prototype.checkGlobals = function (test) {
   }
 };
 
+/**
+ * Fail all tests that are affected by a hook failure.
+ * This is used when the `failHookAffectedTests` option is enabled.
+ *
+ * @private
+ * @param {Suite} suite - The suite containing the affected tests
+ * @param {Error} hookError - The error from the failed hook
+ * @param {string} hookTitle - The title of the failed hook
+ */
+Runner.prototype.failAffectedTests = function (suite, hookError, hookTitle) {
+  if (!this._opts.failHookAffectedTests) {
+    return;
+  }
+
+  var self = this;
+  var errorMessage =
+    'Test skipped due to failure in "' + hookTitle + '": ' + hookError.message;
```

Comment:
Let's try to make this `errorMessage` more global so we don't have to write it multiple times. Maybe it should be a func that takes `hookTitle` and `hookErrorMsg` ?